throttle/contract_tx_limiter_middleware.go

```diff
@@ -127,10 +126,27 @@ func NewContractTxLimiterMiddleware(cfg *ContractTxLimiterConfig,
 		if err := proto.Unmarshal(msg.Data, &msgTx); err != nil {
 			return res, errors.Wrapf(err, "unmarshal call tx %v", msg.Data)
 		}
-
 		if msgTx.VmType != vm.VMType_EVM {
 			return next(state, txBytes, isCheckTx)
 		}
+		if txl.inactiveDeployerContracts == nil ||
+			(txl.contractDataLastUpdated+cfg.ContractDataRefreshInterval) < time.Now().Unix() {
+			ctx, err := createUserDeployerWhitelistCtx(state)
+			if err != nil {
+				return res, errors.Wrap(err, "throttle: context creation")
+			}
+			inactiveDeployerContracts, err := udw.GetInactiveDeployerContracts(ctx)
+			if err != nil {
+				return res, errors.Wrap(err, "throttle: inactiveDeployerContracts creation")
+			}
+			txl.inactiveDeployerContracts = inactiveDeployerContracts
+			// TxLimiter.contractDataLastUpdated will be updated after updating contractToTierMap
+		}
+		contractAddr := loom.UnmarshalAddressPB(msg.To)
+		// contracts which are deployed by deleted deployers should be throttled
+		if txl.inactiveDeployerContracts[contractAddr.String()] {
+			return res, errors.New("contract inactive")
```

Comment:
could use better wording than `contract inactive`

Reply:
vadim suggested "contract inactive" in above comment, should I use "deployer of the contract is deleted" ?

Reply:
yeah, was thinking we need to make it slight verbose.

Reply:
done